test/chplcheck/CaseRules.chpl

```diff
@@ -0,0 +1,41 @@
+module CaseRules {
+    var camelVar: int;
+    const PascalVar: real;
+    var snake_var: string;
+
+    record camelRecord {
+        proc camelCaseMethod() {return 0;}
+        proc PascalCaseMethod() {return 0;}
+        proc snake_case_method() {return 0;}
+    }
+    record PascalRecord {}
+    record snake_record {
+        var camelField, anotherCamelField: int;
+        const camelField2, PascalField: real;
+        var camelField3, snake_field: string;
+    }
+
+    proc camelProc() {return 0;}
+    proc PascalProc() {return 0;}
+    proc snake_proc() {
+        var camelVar: int;
+        const PascalVar: real;
+        var snake_var: string;
+        return 0;
+    }
+
+    proc foo(camelArg: int, PascalArg: string, snake_arg: real) {
+        return camelArg + int(snake_arg) + PascalArg.size();
```

Comment:
This isn't valid Chapel code, is it? `int(snake_arg)`? Perhaps you meant `snake_arg : int`?

Reply:
😅 i'll keep the python in the `.py`! It does build as long as `foo` isn't called, even though the callsite doesn't give any new type info. Something in the resolver? 